tests/integration_test/integration_test.sh

```diff
@@ -376,28 +385,38 @@ function stop_lira_on_error {
 trap "stop_lira_on_error" ERR
 
 # 8. Send in notifications
-printf "\n\nGetting notification token\n"
 
-notification_token=$(docker run -i --rm \
-      -e VAULT_TOKEN=$(cat $vault_token_path) \
-      broadinstitute/dsde-toolbox \
-      vault read -field=notification_token secret/dsde/mint/$env/listener/listener_secret)
+if [ "$use_hmac" == "true" ]; then
+  printf "\n\nGetting hmac key\n"
+  HMAC_KEY=$(docker run -i --rm \
+        -e VAULT_TOKEN=$(cat $vault_token_path) \
+        broadinstitute/dsde-toolbox \
+        vault read -field=current_key secret/dsde/mint/$env/listener/hmac_keys)
+  AUTH_PARAMS="--hmac_key $HMAC_KEY --hmac_key_id current_key"
+else
+  printf "\n\nGetting notification token\n"
+  notification_token=$(docker run -i --rm \
+        -e VAULT_TOKEN=$(cat $vault_token_path) \
+        broadinstitute/dsde-toolbox \
+        vault read -field=notification_token secret/dsde/mint/$env/listener/listener_secret)
+  AUTH_PARAMS="--query_param_token $notification_token"
+fi
 
 printf "\n\nSending in notifications\n"
-# Uses the docker image https://hub.docker.com/r/broadinstitute/python-requests/
+# Uses the docker image built from Dockerfile next to this script
 ss2_workflow_id=$(docker run --rm -v $script_dir:/app \
                     -e LIRA_URL="http://lira:8080/notifications" \
-                    -e NOTIFICATION_TOKEN=$notification_token \
                     -e NOTIFICATION=/app/ss2_notification_dss_${env}.json \
                     --link lira:lira \
-                    broadinstitute/python-requests:3 /app/send_notification.py)
+                    quay.io/humancellatlas/secondary-analysis-mintegration /app/send_notification.py \
+                    $(echo "$AUTH_PARAMS" | xargs))
```

Comment:
If it is using the param token mode, does this mean it will print the token to the console?

Reply:
It won't print to the console. Here's the output when using the query param token:
```
Sending in notifications

ss2_workflow_id: c85cd814-226f-4f78-b47a-15bab9d3f35e
```

Here's the output when using hmac:
```
Sending in notifications

ss2_workflow_id: f4b89dc2-23b9-44bf-8085-f6832e8fe780
```

Reply:
Thanks for the clarification here!